qiskit_nature/optionals.py

```diff
@@ -0,0 +1,90 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Additional optional constants.
+"""
+
+from typing import Optional, Callable
+from shutil import which
+from qiskit.utils import LazyDependencyManager, LazyImportTester, LazySubprocessTester
+
+
+class NatureLazyCommandTester(LazyDependencyManager):
+    """
+    A lazy checker that a command-line tool is available. This just checks if the command is
+    not empty.
+    """
+
+    __slots__ = ("_command",)
```

Comment:
Also, you don't need to do this if you don't want, especially if you'd like to attach data to the objects.  To be honest, I probably shouldn't have done it in Terra to give you the choice of adding arbitrary data - from a Python perspective, it's a bad habit of mine that I over-use `__slots__` (it comes from me preferring stricter typing, where you can't add attributes to stuff).

Reply:
Yes, since I was creating a lazy class I decided to stick to the idiom there but already had someone else asking me about it too :-) I will remove it.